pkg/internal/checkup/checkup.go

```diff
@@ -127,6 +131,34 @@ func (c *Checkup) Results() status.Results {
 	return status.Results{}
 }
 
+func (c *Checkup) waitForVMIToBoot(ctx context.Context) error {
```

Comment:
In the vm latency checkup, we wait for the NIC to have an IP address.
This method is not applicable here, because the NICs done not get an IP address.
When the guest agent connects, the machine had almost fully booted.
@ormergi had shown me a method to make this wait even more precise, but IMO the next thing that should be done is to try and login in a follow-up PR.

I wrote this method, based on other poll logic contained in the code.
It is not copied from kubevirt.

Reply:
ah ok. 
Please add a short explanation on the commit message that you're waiting for the VM to boot by waiting for it to reach VirtualMachineInstanceAgentConnected condition.

Reply:
Done.